src/main.rs

```diff
@@ -60,9 +60,31 @@ fn parse_matches() -> clap::ArgMatches {
                 .takes_value(true)
                 .help("Pulls specific release version. e.g. v1.17.2"),
         )
+        .arg(
+            Arg::with_name("commit")
+                .long("commit")
+                .value_name("HASH")
+                .takes_value(true)
+                .requires("github_user")
+                .requires("repo_name")
+                .help("Pulls specific commit. must be full commit hash. e.g. 8db8e60c48ab064c88a76013597f99c9eb25ed74"),
+        )
+        .arg(
+            Arg::with_name("github_user")
+                .long("github-user")
+                .takes_value(true)
+                .help("Dictates which github user repo the commit is owned by. e.g. gregcusack"),
+        )
+        .arg(
+            Arg::with_name("repo_name")
+                .long("repo-name")
+                .takes_value(true)
+                .default_value("solana")
+                .help("Dictates the repo name to pull the commit from. e.g. agave"),
+        )
```

Comment:
For these two, it's probably better to error if they're not all specified, so I can't do just `--github-user` and get confused later. So could you add the `requires()` for both of these?

Or even neater, an arg group with `requires_all` might be clearest, ie:

```
    .group(ArgGroup::with_name("github_args")
        .args(&["commit", "github_user", "repo_name"])
        .requires_all(&["commit", "github_user", "repo_name"])
        .multiple(true))
```

Note: I haven't tried this, so it might not work. Documentation at https://docs.rs/clap/2.34.0/clap/struct.ArgGroup.html#method.requires_all